src/pages/ni/billing/ToBill.vue

```diff
@@ -239,37 +239,45 @@ export default {
         this.tableLoading = false;
       }
     },
-    async createBills () {
+    async createBillsBatch (shouldBeSent) {
       try {
-        this.$q.dialog({
-          title: 'Confirmation',
-          message: 'Cette opération est définitive. Confirmez-vous ?',
-          ok: 'Oui',
-          cancel: 'Non',
-          options: {
-            type: 'checkbox',
-            model: [true],
-            items: [{ label: 'Envoyer par email', value: true, color: 'primary' }],
+        if (!this.hasSelectedRows) return;
+        const BATCH_SIZE = 50;
+        const promises = [];
+        const bills = this.selected.map(row => ({
+          ...this.$_.omit(row, ['__index']),
+          customerBills: {
+            ...row.customerBills,
+            shouldBeSent: !!shouldBeSent[0],
           },
-        }).onOk(async (shouldBeSent) => {
-          if (!this.hasSelectedRows) return;
-          const bills = this.selected.map(row => ({
-            ...this.$_.omit(row, ['__index']),
-            customerBills: {
-              ...row.customerBills,
-              shouldBeSent: !!shouldBeSent[0],
-            },
-          }));
-          await this.$bills.create({ bills });
-          NotifyPositive('Clients facturés');
-          await this.getDraftBills();
-          this.selected = [];
-        });
+        }));
+
+        for (let i = 0, l = bills.length; i < l; i += BATCH_SIZE) {
+          promises.push(this.$bills.create({ bills: bills.slice(i, i + BATCH_SIZE) }));
+        }
+        await Promise.all(promises);
+        NotifyPositive('Clients facturés');
+        await this.getDraftBills();
+        this.selected = [];
```

Comment:
il faudrait le mettre avant le `await this.getDraftBills();`
et peut etre mettre le Notify en dernier, qu'en penses tu ?

Reply:
tout à fait d'accord !